catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/LocationRepository.java

```diff
@@ -219,9 +222,16 @@ public EntityUpdater getUpdater(Location original, Location updated, boolean pat
   }
 
   private EntityReference getService(Location location) throws IOException {
-    EntityReference ref =
-        EntityUtil.getService(
-            daoCollection.relationshipDAO(), Entity.LOCATION, location.getId(), Entity.STORAGE_SERVICE);
+    EntityReference ref;
+    if (location.getDeleted()) {
```

Comment:
I was thinking about this part of the code. When finding a relationship, we can pass the deleted state of the entity to the relationship lookup. For example, if the location is deleted, then pass Include.DELETED to the query. If the location is not deleted, then pass Include.NON_DELETED to the query. Wdyt?